sklearn/metrics/tests/test_ranking.py

```diff
@@ -881,6 +871,27 @@ def test_binary_clf_curve_implicit_pos_label(curve_func):
         np.testing.assert_allclose(int_curve_part, float_curve_part)
 
 
+# TODO(1.5): Update test to check for error when bytes support is removed.
+@pytest.mark.parametrize("curve_func", [precision_recall_curve, roc_curve])
+@pytest.mark.parametrize("labels_type", ["list", "array"])
+def test_binary_clf_curve_implicit_bytes_pos_label(curve_func, labels_type):
+    # Check that using bytes class labels raises an informative
+    # error for any supported string dtype:
+    labels = _convert_container([b"a", b"b"], labels_type)
+    msg = (
+        "y_true takes value in {b'a', b'b'} and pos_label is not "
+        "specified: either make y_true take value in {0, 1} or "
+        "{-1, 1} or pass pos_label explicitly."
+    )
+    with warnings.catch_warnings(record=True) as record:
```

Comment:
The CI is still unhappy with this. At this point, I think it's okay to ignore the error:

```patch
diff --git a/sklearn/metrics/tests/test_ranking.py b/sklearn/metrics/tests/test_ranking.py
index e5e8856e21..fe7ad24b48 100644
--- a/sklearn/metrics/tests/test_ranking.py
+++ b/sklearn/metrics/tests/test_ranking.py
@@ -19,6 +19,7 @@ from sklearn.utils._testing import assert_allclose
 from sklearn.utils._testing import assert_almost_equal
 from sklearn.utils._testing import assert_array_equal
 from sklearn.utils._testing import assert_array_almost_equal
+from sklearn.utils._testing import ignore_warnings
 
 from sklearn.metrics import accuracy_score
 from sklearn.metrics import auc
@@ -872,6 +873,7 @@ def test_binary_clf_curve_implicit_pos_label(curve_func):
 
 
 # TODO(1.5): Update test to check for error when bytes support is removed.
+@ignore_warnings(category=FutureWarning)
 @pytest.mark.parametrize("curve_func", [precision_recall_curve, roc_curve])
 @pytest.mark.parametrize("labels_type", ["list", "array"])
 def test_binary_clf_curve_implicit_bytes_pos_label(curve_func, labels_type):
@@ -883,13 +885,8 @@ def test_binary_clf_curve_implicit_bytes_pos_label(curve_func, labels_type):
         "specified: either make y_true take value in {0, 1} or "
         "{-1, 1} or pass pos_label explicitly."
     )
-    with warnings.catch_warnings(record=True) as record:
-        with pytest.raises(ValueError, match=msg):
-            curve_func(labels, [0.0, 1.0])
-
-        assert str(record[0].message).startswith(
-            "Support for labels represented as bytes is deprecated"
-        )
+    with pytest.raises(ValueError, match=msg):
+        curve_func(labels, [0.0, 1.0])
 
 
 @pytest.mark.parametrize("curve_func", CURVE_FUNCS)
```